### DISCUSSION TITLE
isDisplayed() returning false, even if the element is present and visible

### DISCUSSION THREAD

**Discussion by Nerwen18** [selenide/selenide#2873] [Q&A]
Hello everyone, I'm having the following HTML elements:

```html
<form id="formId">
    <div id="divA" class="error message field">
        <span class="m error icon"></span>
        <span>textA</span>
    </div>
    <div id="divB" class="error message field">
        <span class="m error icon"></span>
        <span>textB</span>
    </div>
</form>
```

and i'm checking the visibility for both elements, by ids `divA` and `divB`. Both these elements belong to the same frame, so I switched to it and then made the asserts. The first one is visible, but the second assert returns false. As you can see, it doesn't have `style="display:none"`, is not hidden, is visible on the screen. 

Can you advise any other checks or a workaround? Thank you!

**Comment by asolntsev** [comment]
@Nerwen18 Hardly anyone can help you without seeing the whole html page, and your code for asserting.
Please provide a minimal reproducible example that we can run and see the problem.

When I just run your code, it work without errors:

```java
    openFile("test_page.html");
    $("#divA").shouldBe(visible);  // divA is visible
    $("#divB").shouldBe(visible);  // divB is visible
```

where file `test_page.html` contains:

```html
<!DOCTYPE html>
<html>
<head>
  <title>Test page</title>
</head>
<body>

<form id="formId">
  <div id="divA" class="error message field">
    <span class="m error icon"></span>
    <span>textA</span>
  </div>
  <div id="divB" class="error message field">
    <span class="m error icon"></span>
    <span>textB</span>
  </div>
</form>

</body>
</html>
```
